app/controllers/application_controller.rb

```diff
@@ -1,14 +1,34 @@
 class ApplicationController < ActionController::Base
+  
   protect_from_forgery with: :exception
   include SessionsHelper
-  
   helper_method :current_order
 
   def current_order
-    if !session[:order_id].nil?
-      Order.find(session[:order_id])
+    if !session[:order_id].nil? && !search_order.nil?
+      order = Order.find(session[:order_id])
+      if order.user_id != current_user.id
+        create_order
+        session[:order_id] = @o.id
+        return @o
+      else
+        return order
+      end
     else
-      Order.new
+      create_order
+      session[:order_id] = @o.id
+      return @o
     end
+
   end
+
+  private
+    def search_order
```

Comment:
Shouldn't this be called `current_order` or something? This name is not that good

Reply:
I have another function called current_order